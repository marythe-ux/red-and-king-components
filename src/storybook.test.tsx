import initStoryshots from '@storybook/addon-storyshots';
jest.mock('./styles/GlobalStyles', () => 'GlobalStyles');
initStoryshots();
