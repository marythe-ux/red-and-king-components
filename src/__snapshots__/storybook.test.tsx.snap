// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Button Default 1`] = `
Array [
  <GlobalStyles />,
  <button
    className="css-yi23fg"
    color="primary"
    type="button"
  >
    Label
  </button>,
]
`;

exports[`Storyshots Components/Button Large 1`] = `
Array [
  <GlobalStyles />,
  <button
    className="css-1ecuqcw"
    color="primary"
    size="large"
    type="button"
  >
    Label
  </button>,
]
`;

exports[`Storyshots Components/Button Small 1`] = `
Array [
  <GlobalStyles />,
  <button
    className="css-1qxsimw"
    color="primary"
    size="small"
    type="button"
  >
    Label
  </button>,
]
`;

exports[`Storyshots Design tokens/Colors Colors 1`] = `
Array [
  <GlobalStyles />,
  <h1>
    Design tokens colors
  </h1>,
  <div
    className="css-1nndm63"
  >
    <p
      className="title"
    >
      White
    </p>
    <div
      className="container"
    >
      <span
        className="css-6x4xmh"
        color="#ffffff"
      >
        White
        <br />
        #ffffff
      </span>
    </div>
  </div>,
  <div
    className="css-1nndm63"
  >
    <p
      className="title"
    >
      Dark
    </p>
    <div
      className="container"
    >
      <span
        className="css-1qn4u0p"
        color="#000000"
      >
        Dark
        <br />
        #000000
      </span>
    </div>
  </div>,
  <div
    className="css-1nndm63"
  >
    <p
      className="title"
    >
      primary
    </p>
    <div
      className="container"
    >
      <span
        className="css-y6gnpf"
        color="#f5e6fc"
      >
        50
        <br />
        #f5e6fc
      </span>
      <span
        className="css-wgyev"
        color="#e6bff9"
      >
        100
        <br />
        #e6bff9
      </span>
      <span
        className="css-18t54z7"
        color="#d694f6"
      >
        200
        <br />
        #d694f6
      </span>
      <span
        className="css-6f4yfg"
        color="#c566f3"
      >
        300
        <br />
        #c566f3
      </span>
      <span
        className="css-1vmzfxk"
        color="#b73ef0"
      >
        400
        <br />
        #b73ef0
      </span>
      <span
        className="css-8df64y"
        color="#A800EB"
      >
        500
        <br />
        #A800EB
      </span>
      <span
        className="css-111wpt6"
        color="#9607e5"
      >
        600
        <br />
        #9607e5
      </span>
      <span
        className="css-1wesx5n"
        color="#7c0ade"
      >
        700
        <br />
        #7c0ade
      </span>
      <span
        className="css-13hzm5x"
        color="#630bd9"
      >
        800
        <br />
        #630bd9
      </span>
      <span
        className="css-1ur9nty"
        color="#150dd2"
      >
        900
        <br />
        #150dd2
      </span>
    </div>
  </div>,
  <div
    className="css-1nndm63"
  >
    <p
      className="title"
    >
      semantic
    </p>
    <div
      className="container"
    >
      <span
        className="css-2mfgsz"
        color="#ff6000"
      >
        danger
        <br />
        #ff6000
      </span>
    </div>
  </div>,
]
`;

exports[`Storyshots Design tokens/Spacing Spacing 1`] = `
Array [
  <GlobalStyles />,
  <h1>
    Design tokens spacing
  </h1>,
  <div
    className="css-630khx"
  >
    <p
      className="title"
    >
      xxs
    </p>
    <p
      className="space"
    />
  </div>,
  <div
    className="css-plio13"
  >
    <p
      className="title"
    >
      xs
    </p>
    <p
      className="space"
    />
  </div>,
  <div
    className="css-bcf2m"
  >
    <p
      className="title"
    >
      s
    </p>
    <p
      className="space"
    />
  </div>,
  <div
    className="css-14it4ly"
  >
    <p
      className="title"
    >
      m
    </p>
    <p
      className="space"
    />
  </div>,
  <div
    className="css-fm0ppo"
  >
    <p
      className="title"
    >
      base
    </p>
    <p
      className="space"
    />
  </div>,
  <div
    className="css-zjesl3"
  >
    <p
      className="title"
    >
      l
    </p>
    <p
      className="space"
    />
  </div>,
  <div
    className="css-1mpunvi"
  >
    <p
      className="title"
    >
      xl
    </p>
    <p
      className="space"
    />
  </div>,
  <div
    className="css-1psdnd9"
  >
    <p
      className="title"
    >
      xxl
    </p>
    <p
      className="space"
    />
  </div>,
  <div
    className="css-cb5u1f"
  >
    <p
      className="title"
    >
      xxxl
    </p>
    <p
      className="space"
    />
  </div>,
  <div
    className="css-pm9xa5"
  >
    <p
      className="title"
    >
      xxxxl
    </p>
    <p
      className="space"
    />
  </div>,
]
`;

exports[`Storyshots Example/Header Logged In 1`] = `
Array [
  <GlobalStyles />,
  <header>
    <div
      className="wrapper"
    >
      <div>
        <svg
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z"
              fill="#FFF"
            />
            <path
              d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z"
              fill="#555AB9"
            />
            <path
              d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z"
              fill="#91BAF8"
            />
          </g>
        </svg>
        <h1>
          Acme
        </h1>
      </div>
      <div>
        <span
          className="welcome"
        >
          Welcome, 
          <b>
            Jane Doe
          </b>
          !
        </span>
        <button
          className="css-1qxsimw"
          color="secondary"
          size="small"
          type="button"
        >
          Log out
        </button>
      </div>
    </div>
  </header>,
]
`;

exports[`Storyshots Example/Header Logged Out 1`] = `
Array [
  <GlobalStyles />,
  <header>
    <div
      className="wrapper"
    >
      <div>
        <svg
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z"
              fill="#FFF"
            />
            <path
              d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z"
              fill="#555AB9"
            />
            <path
              d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z"
              fill="#91BAF8"
            />
          </g>
        </svg>
        <h1>
          Acme
        </h1>
      </div>
      <div>
        <button
          className="css-1qxsimw"
          color="secondary"
          size="small"
          type="button"
        >
          Log in
        </button>
        <button
          className="css-1qxsimw"
          size="small"
          type="button"
        >
          Sign up
        </button>
      </div>
    </div>
  </header>,
]
`;

exports[`Storyshots Example/Page Logged In 1`] = `
Array [
  <GlobalStyles />,
  <article>
    <header>
      <div
        className="wrapper"
      >
        <div>
          <svg
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fillRule="evenodd"
            >
              <path
                d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z"
                fill="#FFF"
              />
              <path
                d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z"
                fill="#555AB9"
              />
              <path
                d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z"
                fill="#91BAF8"
              />
            </g>
          </svg>
          <h1>
            Acme
          </h1>
        </div>
        <div>
          <button
            className="css-1qxsimw"
            color="secondary"
            onClick={[Function]}
            size="small"
            type="button"
          >
            Log in
          </button>
          <button
            className="css-1qxsimw"
            onClick={[Function]}
            size="small"
            type="button"
          >
            Sign up
          </button>
        </div>
      </div>
    </header>
    <section>
      <h2>
        Pages in Storybook
      </h2>
      <p>
        We recommend building UIs with a
         
        <a
          href="https://componentdriven.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>
            component-driven
          </strong>
        </a>
         
        process starting with atomic components and ending with pages.
      </p>
      <p>
        Render pages with mock data. This makes it easy to build and review page states without needing to navigate to them in your app. Here are some handy patterns for managing page data in Storybook:
      </p>
      <ul>
        <li>
          Use a higher-level connected component. Storybook helps you compose such data from the "args" of child component stories
        </li>
        <li>
          Assemble data in the page component from your services. You can mock these services out using Storybook.
        </li>
      </ul>
      <p>
        Get a guided tutorial on component-driven development at
         
        <a
          href="https://storybook.js.org/tutorials/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Storybook tutorials
        </a>
        . Read more in the
         
        <a
          href="https://storybook.js.org/docs"
          rel="noopener noreferrer"
          target="_blank"
        >
          docs
        </a>
        .
      </p>
      <div
        className="tip-wrapper"
      >
        <span
          className="tip"
        >
          Tip
        </span>
         Adjust the width of the canvas with the
         
        <svg
          height="10"
          viewBox="0 0 12 12"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M1.5 5.2h4.8c.3 0 .5.2.5.4v5.1c-.1.2-.3.3-.4.3H1.4a.5.5 0 01-.5-.4V5.7c0-.3.2-.5.5-.5zm0-2.1h6.9c.3 0 .5.2.5.4v7a.5.5 0 01-1 0V4H1.5a.5.5 0 010-1zm0-2.1h9c.3 0 .5.2.5.4v9.1a.5.5 0 01-1 0V2H1.5a.5.5 0 010-1zm4.3 5.2H2V10h3.8V6.2z"
              fill="#999"
              id="a"
            />
          </g>
        </svg>
        Viewports addon in the toolbar
      </div>
    </section>
  </article>,
]
`;

exports[`Storyshots Example/Page Logged Out 1`] = `
Array [
  <GlobalStyles />,
  <article>
    <header>
      <div
        className="wrapper"
      >
        <div>
          <svg
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fillRule="evenodd"
            >
              <path
                d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z"
                fill="#FFF"
              />
              <path
                d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z"
                fill="#555AB9"
              />
              <path
                d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z"
                fill="#91BAF8"
              />
            </g>
          </svg>
          <h1>
            Acme
          </h1>
        </div>
        <div>
          <button
            className="css-1qxsimw"
            color="secondary"
            onClick={[Function]}
            size="small"
            type="button"
          >
            Log in
          </button>
          <button
            className="css-1qxsimw"
            onClick={[Function]}
            size="small"
            type="button"
          >
            Sign up
          </button>
        </div>
      </div>
    </header>
    <section>
      <h2>
        Pages in Storybook
      </h2>
      <p>
        We recommend building UIs with a
         
        <a
          href="https://componentdriven.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>
            component-driven
          </strong>
        </a>
         
        process starting with atomic components and ending with pages.
      </p>
      <p>
        Render pages with mock data. This makes it easy to build and review page states without needing to navigate to them in your app. Here are some handy patterns for managing page data in Storybook:
      </p>
      <ul>
        <li>
          Use a higher-level connected component. Storybook helps you compose such data from the "args" of child component stories
        </li>
        <li>
          Assemble data in the page component from your services. You can mock these services out using Storybook.
        </li>
      </ul>
      <p>
        Get a guided tutorial on component-driven development at
         
        <a
          href="https://storybook.js.org/tutorials/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Storybook tutorials
        </a>
        . Read more in the
         
        <a
          href="https://storybook.js.org/docs"
          rel="noopener noreferrer"
          target="_blank"
        >
          docs
        </a>
        .
      </p>
      <div
        className="tip-wrapper"
      >
        <span
          className="tip"
        >
          Tip
        </span>
         Adjust the width of the canvas with the
         
        <svg
          height="10"
          viewBox="0 0 12 12"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M1.5 5.2h4.8c.3 0 .5.2.5.4v5.1c-.1.2-.3.3-.4.3H1.4a.5.5 0 01-.5-.4V5.7c0-.3.2-.5.5-.5zm0-2.1h6.9c.3 0 .5.2.5.4v7a.5.5 0 01-1 0V4H1.5a.5.5 0 010-1zm0-2.1h9c.3 0 .5.2.5.4v9.1a.5.5 0 01-1 0V2H1.5a.5.5 0 010-1zm4.3 5.2H2V10h3.8V6.2z"
              fill="#999"
              id="a"
            />
          </g>
        </svg>
        Viewports addon in the toolbar
      </div>
    </section>
  </article>,
]
`;
